src/sentry/static/sentry/app/utils/tokenizeSearch.tsx

```diff
@@ -28,24 +30,28 @@ export function tokenizeSearch(query: string) {
   };
 
   for (const token of tokens) {
-    const tokenChars = Array.from(token);
     let tokenState: 'query' | 'tags' = 'query';
 
-    tokenChars.forEach((char, idx) => {
-      const nextChar = tokenChars.length > idx ? tokenChars[idx + 1] : null;
+    // Traverse the token and determine if it is a tag
+    // condition or bare words.
+    for (let i = 0, len = token.length; i < len; i++) {
+      const char = token[i];
 
-      if (idx === 0 && [':', ' '].includes(char)) {
-        return;
+      if (i === 0 && char === '"') {
```

Comment:
```suggestion
      if (i === 0 && (char === '"' || char === ':')) {
```

We also want to bail out on tokens that begin with a colon. These tokens are still queries.

Test case:
```js
      {
        name: 'tokens that begin with a colon are still queries',
        string: 'country:canada :unresolved',
        object: {query: [':unresolved'], country: ['canada']},
      },
```

Reply:
Fixed now in both the JS and python implementation.

Reply:
Oh interesting. I thought the Python covered this case. I had searched with a query that began with a colon on SaaS .